### ISSUE TITLE
Crash on file rename or move

### ISSUE THREAD

**Issue by wjagodfrey** [godotengine/godot#82652]
### Godot version

4.2-rc5

### System information

MacOS 12.3

### Issue description

When renaming or moving a file in-editor, Godot crashes. Happens 99.9% of the time.

Video of the issue [here](https://youtu.be/gjQlb6y6ysU).

### Steps to reproduce

1. Load a project
2. Move a directory containing files that are used by other scenes
3. Wait for the crash

### Minimal reproduction project

I don't think an MRP is necessary, but if you can't replicate I'll give it a go.

**Comment by Calinou** [comment]
Can you reproduce this in 4.1.1, or only in 4.2.dev5? If you can't reproduce this in 4.1.1, please also test 4.2.dev4 and older dev builds.

**Comment by wjagodfrey** [comment]
Neither `4.1.1` or `4.2.dev4`  have the crash

**Comment by wjagodfrey** [comment]
Moving more than one directory in `4.2.dev4` did result in the same crash, however the problem seems worse in `dev5`

**Comment by Jordyfel** [comment]
Should be fixed by #81725

**Comment by akien-mga** [comment]
Please test in the upcoming 4.2-dev6 (should be released tomorrow most likely) and confirm.

**Comment by wjagodfrey** [comment]
Crash is still immediately reproducible in 4.2-dev6. Sorry guys

**Comment by Rindbee** [comment]
If I close all scenes (or scenes that use these resource files), will it crash?

**Comment by wjagodfrey** [comment]
Yes, that seems to be the case. If all scenes are closed there is no crash. There's no crash log in the user data directory or elsewhere that I can find.

**Comment by Rindbee** [comment]
The post-crash popup may contain useful information.

**Comment by wjagodfrey** [comment]
Ah I'd posted one on a similar issue and someone told me it wasn't helpful. You can find my system crashlog [here](https://gist.github.com/wjagodfrey/9cc091f103f178093d09dd328eb9e9e0).

**Comment by Rindbee** [comment]
I reproduced a crash, but I'm not sure if it's the same one.

Steps to reproduce:
1. Create a new project;
2. Add a glb file (`cube.glb`) to the project directory;
3. Create 3 directories `a`, `b`, `c`
4. Create a 3D scene, use `cube.glb` to instantiate the sub-scene, save the scene (`cube.tscn`) and close it.
5. Move `cube.glb` and `cube.tscn` to directory a.
6. Create another 3D scene, use `cube.tscn` to instantiate the sub-scene, save the scene (`node_3d.tscn`), and open `cube.tscn` again. (The order of open scene tabs seems to be the key.)
7. Moved directory `a` to `b`, no errors, no crashes.
8. Then move the directory `a` from `b` to `c`, output an error, and then crash.

```
ERROR: Cannot open file 'res://b/a/cube.tscn'.
   at: load (scene/resources/resource_format_text.cpp:1641)
ERROR: Failed loading resource: res://b/a/cube.tscn. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:275)
ERROR: Method/function failed. Returning: err
   at: pack (scene/resources/packed_scene.cpp:1077)
ERROR: Condition "err != OK" is true. Returning: false
   at: check_and_update_scene (editor/editor_data.cpp:700)
```

```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.2.dev.custom_build (f5696c311cdb09e0a34fa4ba7ef5d2524c515b89)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7fb71a042520] (??:0)
[2] /opt/godot/godot-editor-l10n/godot/bin/godot.linuxbsd.editor.dev.x86_64.llvm(__dynamic_cast+0x2d) [0x556ff886c50d] (??:?)
[3] Node* Object::cast_to<Node>(Object*) (/opt/godot/godot-editor-l10n/godot/./core/object/object.h:792)
[4] VariantObjectClassChecker<Node*>::check(Variant const&) (/opt/godot/godot-editor-l10n/godot/./core/variant/binder_common.h:230)
[5] VariantCasterAndValidate<Node*>::cast(Variant const**, unsigned int, Callable::CallError&) (/opt/godot/godot-editor-l10n/godot/./core/variant/binder_common.h:255)
[6] void call_with_variant_args_helper<__UnexistingClass, Dictionary, Node*, 0ul, 1ul>(__UnexistingClass*, void (__UnexistingClass::*)(Dictionary, Node*), Variant const**, Callable::CallError&, IndexSequence<0ul, 1ul>) (/opt/godot/godot-editor-l10n/godot/./core/variant/binder_common.h:303)
[7] void call_with_variant_args_dv<__UnexistingClass, Dictionary, Node*>(__UnexistingClass*, void (__UnexistingClass::*)(Dictionary, Node*), Variant const**, int, Callable::CallError&, Vector<Variant> const&) (/opt/godot/godot-editor-l10n/godot/./core/variant/binder_common.h:451)
[8] MethodBindT<Dictionary, Node*>::call(Object*, Variant const**, int, Callable::CallError&) const (/opt/godot/godot-editor-l10n/godot/./core/object/method_bind.h:335)
[9] Object::callp(StringName const&, Variant const**, int, Callable::CallError&) (/opt/godot/godot-editor-l10n/godot/core/object/object.cpp:766)
[10] Callable::callp(Variant const**, int, Variant&, Callable::CallError&) const (/opt/godot/godot-editor-l10n/godot/core/variant/callable.cpp:62)
[11] CallQueue::_call_function(Callable const&, Variant const*, int, bool) (/opt/godot/godot-editor-l10n/godot/core/object/message_queue.cpp:220)
[12] CallQueue::flush() (/opt/godot/godot-editor-l10n/godot/core/object/message_queue.cpp:326)
[13] SceneTree::physics_process(double) (/opt/godot/godot-editor-l10n/godot/scene/main/scene_tree.cpp:471)
[14] Main::iteration() (/opt/godot/godot-editor-l10n/godot/main/main.cpp:3545)
[15] OS_LinuxBSD::run() (/opt/godot/godot-editor-l10n/godot/platform/linuxbsd/os_linuxbsd.cpp:933)
[16] /opt/godot/godot-editor-l10n/godot/bin/godot.linuxbsd.editor.dev.x86_64.llvm(main+0x1a0) [0x556ff3e24890] (/opt/godot/godot-editor-l10n/godot/platform/linuxbsd/godot_linuxbsd.cpp:76)
[17] /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7fb71a029d90] (??:0)
[18] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7fb71a029e40] (??:0)
[19] /opt/godot/godot-editor-l10n/godot/bin/godot.linuxbsd.editor.dev.x86_64.llvm(_start+0x25) [0x556ff3e24625] (??:?)
-- END OF BACKTRACE --
================================================================
```
Open the project again and perform the above operations. You will only get errors and no crashes.

**Comment by wjagodfrey** [comment]
| After the crash, the files were successfully moved, but the path of source_file in cube.glb.import was not updated. (Resources in other scene files have been updated to new paths.)

This does sound like what I'm seeing, though I suppose it's possible I'm experiencing two separate crashes.

It's a long shot, but is it possible this is related to https://github.com/godotengine/godot/issues/82422?

**Comment by valkyrienyanko** [comment]
This issue happens 80% of the time when I move C# scripts from one folder to another. Although the editor does not crash, scenes fail to open and the game won't run anymore. I find I have to move one script, test the game to see if it still works, if it does work then commit these changes through git. Then repeat the process for the next script. Very tedious..

For example I move `Player.cs` script from folder A to folder B. `player_ship.tscn` fails to update with the new path for the player script. Which means I can't open this scene anymore unless I go into a text editor and manually edit the path. Tested on 4.1.1-stable. Have yet to test on 4.2-dev-x build.

**Comment by wjagodfrey** [comment]
Can replicate in 4.2-beta1. Video [here](https://www.youtube.com/watch?v=g4spXOssxPU). Possibly related to another .glb import crash I've reported here https://github.com/godotengine/godot/issues/82398

Also bumped into something else while replicating that may be related. Moving a file by dragging - no modifier keys pressed - ended up copying it, and trying to overwrite one with the other led to a crash. Video [here](https://www.youtube.com/watch?v=JX-MyCCwR7g).

**Comment by AvikArefin** [comment]
I am not sure if this is related, but on 4.1.2 stable, sometimes I can't rename a file.
Minimal steps to reproduce: work on a scene that's saved as .tscn. then close it in the editor [or keep it]. Try to rename the scene file. It won't edit the name.

Hacky way to fix it:
Select another file by mouse left click. then select the previous file again.
Now you can edit the name.

**Comment by GigglingGalaxy** [comment]
Here is minimal repro...never done git before so hope this is right/okay. [Min Repro Rename Bug](https://github.com/GigglingGalaxy/Godot42RenameBugRepro)

**Comment by Jordyfel** [comment]
@GigglingGalaxy Your bug seems different, please test on 4.3 dev 5 and if it is still present, open a new issue

**Comment by fdstevex** [comment]
Adding some info here - ran into this issue and figured out that we had deleted the Git plugin, but not changed the setting that references it.

```
version_control/plugin_name="GitPlugin"
```

With that line in, but no GitPlugin add-on, the editor would crash renaming files. Once we removed this, the crashes stopped.

**Comment by jitdev87** [comment]
Still happening in 4.7.2.stable. No errors in the log or anything. I can reproduce it consistently when trying to rename a TextureButton node.

`Generated 'ID:-9223369974179876709' preview in 15598 usec
Accessibility: window 1 adapter created.
Accessibility: window 1 adapter initial update completed.
Accessibility: window 1 adapter deactivation completed.
Accessibility: window 1 adapter destroyed.
Accessibility: window 2 adapter created.
Accessibility: window 2 adapter activated.
Accessibility: window 2 adapter initial update completed.
Accessibility: window 2 adapter initial update completed.`
